ibis/backends/conftest.py

```diff
@@ -38,72 +40,111 @@ def data_directory() -> Path:
     )
 
 
-def _random_identifier(suffix):
-    return f'__ibis_test_{suffix}_{util.guid()}'
+def _random_identifier(suffix: str) -> str:
+    return f"__ibis_test_{suffix}_{util.guid()}"
 
 
-def _get_all_backends() -> List[str]:
-    """
-    Return the list of known backend names.
+@lru_cache(maxsize=None)
+def _get_all_backends() -> frozenset[str]:
+    """Return the set of known backend names.
+
+    Notes
+    -----
+    This function returns a frozenset to prevent cache pollution.
+
+    If a `set` is used, then any in-place modifications to the set
+    are visible to every caller of this function.
     """
-    return [
+    return frozenset(
         entry_point.name
         for entry_point in importlib_metadata.entry_points()["ibis.backends"]
-    ]
+    )
 
 
 def _backend_name_to_class(backend_str: str):
```

Comment:
Can we rewrite this to `_load_backend[_or_skip](name, data_directory)`?

Reply:
Changed to `_get_backend_conf`. I can do the `data_directory` move in a follow up.